### PR TITLE
fix(test): Values check test should account for `IgnoreInTests` column option

### PR DESCRIPTION
**Pull request by candiduslynx**

### CODE REVIEW

**Review by hermanschaaf — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

File: plugins/source_testing.go [hunk 1]
```diff
@@ -107,8 +107,15 @@ func validateResources(t *testing.T, resources []*schema.Resource) {
...
-		if !hasValue && !(table.Columns[i].Name == "_cq_parent_id" && table.Parent == nil) {
-			t.Errorf("table: %s column %s has no values", table.Name, table.Columns[i].Name)
+		col := table.Columns[i]
+		switch {
+		case col.IgnoreInTests,
+			hasValue,
+			col.Name == "_cq_parent_id" && table.Parent != nil:
+		// nop
+		default:
+			return fmt.Errorf("table: %s column %s has no values", table.Name, table.Columns[i].Name)
```
Comment (yevgenypats) [inline plugins/source_testing.go:117]: I don't think it should return an error but rather call `t.Errorf`

Comment (yevgenypats) [reply plugins/source_testing.go:117]: Also I don't think a switch case is needed here but better just a simple `if`

**Review by yevgenypats — CHANGES_REQUESTED** [review]
almost there.

File: plugins/source_testing.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source_testing.go:117]: > call `t.Errorf`
It makes impossible creating a negative test

> simple `if`
The `if` clause started to look too overloaded

File: plugins/source_testing.go [hunk 2]
```diff
@@ -107,8 +116,15 @@ func validateResources(t *testing.T, resources []*schema.Resource) {
...
-			t.Errorf("table: %s column %s has no values", table.Name, table.Columns[i].Name)
+		col := table.Columns[i]
+		switch {
+		case col.IgnoreInTests,
+			hasValue,
+			col.Name == "_cq_parent_id" && table.Parent != nil:
+		// nop
+		default:
+			errs = append(errs, fmt.Errorf("table: %s column %s has no values", table.Name, table.Columns[i].Name))
 		}
```
Comment (hermanschaaf) [inline plugins/source_testing.go:127]:
```suggestion
		emptyExpected := col.Name == "_cq_parent_id" && table.Parent == nil
		if !col.IgnoreInTests && !emptyExpected && !hasValue {
		errs = append(errs, fmt.Errorf("table: %s column %s has no values", table.Name, table.Columns[i].Name))
		}
```

Maybe this ☝️  (I think this also fixes a bug related to the parent columns being null?)

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Few issues:
* no need to use `testify` package
* no need to change function signature. just call `t.Errorf` directly instead of passing errors around.
* use simple `if else` instead of switch

**Review by yevgenypats — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**